### DISCUSSION TITLE
Add support for HarmonyOS NEXT, publish to Huawei Appgalley

### DISCUSSION THREAD

**Discussion by Aoan2011** [breezy-weather/breezy-weather#1243] [Ideas]
### Describe your idea

I am a Chinese. This weather app is fantastic! but my phone is going to upgrade to HarmonyOS NEXT and this weather app is going to be uninstalled automatically, i want to use this app and i want to upgrade too. Could you make a support for it? Please.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.6 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Someone already asked but was unable to explain what was missing exactly to be able to install on Huawei phones.

The app is available to download as an APK from GitHub releases, and you should be able to install it on any Android phone.

**Reply by Aoan2011** [reply]
> Someone already asked but was unable to explain what was missing exactly to be able to install on Huawei phones.
> 
> The app is available to download as an APK from GitHub releases, and you should be able to install it on any Android phone.

actually，HUAWEI phones can run the app with the version harmonyos 4.2 which is powered by Android , but in harmongos next, it is Huawei's own kernel so that it can't run Android apps or even opening the.apk file
I want  to use it this weather app is really good and i am looking forward  to  looking  this  app  on  Huawei appgalley

**Reply by Aoan2011** [reply]
and most of the Android apps in china is support

**Reply by papjul** [reply]
> in harmongos next, it is Huawei's own kernel so that it can't run Android apps or even opening the.apk file

Then, there's nothing we can do? If you want to run an Android app on Windows, you will need an Android emulator. Try looking for an Android emulator compatible with HarmonyOS NEXT?

**Reply by Aoan2011** [reply]
sure，but I want this app compatible  with harmonyos  NEXT by not using an Android  emulator

**Reply by papjul** [reply]
It's not technically possible, this app is written with the Android SDK. If there is no Android, then it cannot work. It's like asking if you can run an APK on iOS, you can't ;)

**Comment by Aoan2011** [comment]
So, would youl like to develop a version for HarmonyOS NEXT?

**Reply by papjul** [reply]
I have no plans to develop an app for a proprietary system I don't use. You can do your own app but please use a different name than "Breezy Weather", and make sure any algorithm/design you re-use is also LGPL-3.0 licensed ;)

**Reply by Aoan2011** [reply]
ok，thanks
